docs/data/charts/legend/CustomLegend.tsx

```diff
@@ -5,39 +5,30 @@ import Typography from '@mui/material/Typography';
 import { useTheme } from '@mui/material/styles';
 import { useLegend } from '@mui/x-charts/hooks';
 import { LineChart } from '@mui/x-charts/LineChart';
+import {
+  ChartsLabelCustomMarkProps,
+  ChartsLabelMark,
+} from '@mui/x-charts/ChartsLabel';
 
-type IconProps = {
-  color: string;
-  size: number;
-};
-
-function LineWithMark({ color, size }: IconProps) {
+function LineWithMark({ color, ...props }: ChartsLabelCustomMarkProps) {
   return (
     <svg
       viewBox="0 0 24 24"
-      width={size}
-      height={size}
       stroke={color}
       strokeWidth={3}
       strokeLinecap="round"
       fill="none"
+      {...props}
     >
       <path d="M 1 12.5 L 7 12.5 M 17 12.5 L 23 12.5" />
       <circle cx={12} cy={12.5} r={5} />
     </svg>
   );
 }
 
-function DashedLine({ color, size }: IconProps) {
+function DashedLine({ color, ...props }: ChartsLabelCustomMarkProps) {
```

Comment:
Is it a good pattern to spread all props? If we add more props that aren't meant to be spread onto the DOM, won't we be setting attributes that don't exist? IMO, that's a benefit of `ownerState`, but we seem to be deviating from it.